### PR TITLE
Issues/200 - Issues with nested Any in events

### PR DESCRIPTION
**Pull request by meywood**

### CODE REVIEW

File: src/main/java/com/casper/sdk/model/clvalue/cltype/CLTypeMap.java [hunk 1]
```diff
@@ -29,6 +29,25 @@ public void setKeyValueTypes(CLTypeMapEntryType keyValueTypes) {
...
     private CLTypeMapEntryType keyValueTypes;
 
+    @Override
+    public boolean isDeserializable() {
+
+        if (getKeyValueTypes().keyType instanceof CLTypeAny || getKeyValueTypes().valueType instanceof CLTypeAny) {
+            // The map contains an 'Any' type therefore cannot be deserialized
+            return false;
+        } else if (getKeyValueTypes().valueType instanceof AbstractCLTypeWithChildren) {
+           return getChildTypes().stream().allMatch(childType -> {
```
Comment (stormeye2000) [inline src/main/java/com/casper/sdk/model/clvalue/cltype/CLTypeMap.java:39]: extract into another method or class

File: src/main/java/com/casper/sdk/model/clvalue/cltype/CLTypeList.java [hunk 2]
```diff
@@ -50,4 +50,16 @@ public void setListType(AbstractCLType listType) {
         getChildTypes().clear();
         getChildTypes().add(listType);
     }
+
+    @Override
+    public boolean isDeserializable() {
+
+        return getChildTypes().stream().allMatch(childType -> {
+            if (childType instanceof CLTypeAny) {
```
Comment (stormeye2000) [inline src/main/java/com/casper/sdk/model/clvalue/cltype/CLTypeList.java:58]: extract into another method or class

File: src/main/java/com/casper/sdk/model/clvalue/cltype/CLTypeMap.java (hunk 1 above)
Comment (meywood) [reply src/main/java/com/casper/sdk/model/clvalue/cltype/CLTypeMap.java:39]: The list and map have slightly different behaviour. Personally I find DRY for two similar methods is overkill. If there were 3 instance of this I'd consider it

File: src/main/java/com/casper/sdk/model/clvalue/cltype/CLTypeList.java (hunk 2 above)
Comment (meywood) [reply src/main/java/com/casper/sdk/model/clvalue/cltype/CLTypeList.java:58]: See previous comment

**Review by stormeye2000 — APPROVED** [review]

**Review by cnorburn — APPROVED** [review]

**Pull request merged**